ui/library/src/utils/scroll.ts

```diff
@@ -78,13 +80,54 @@ export function scrollToPosition(
     leftPixels = (width * (PDF_HEIGHT_POINTS - bottomPoints)) / PDF_HEIGHT_POINTS;
   }
 
-  targetDiv.scrollTo({
-    top: Math.floor(heightWithMargins * pageIndex + marginTopPixels + (height - bottomPixels)),
+  // Find page element
+  const pageId = generatePageIdFromIndex(pageIndex);
+  const pageIdElement = document.getElementById(pageId);
+  if (!pageIdElement) {
+    return;
+  }
+
+  // Find first scrollable parent
+  const parentElement = getScrollParent(pageIdElement);
+  if (!parentElement) {
+    return;
+  }
+
+  const objectForCalculateTopPx: ObjectForCalculateTopPx = {
+    heightWithMargins: heightWithMargins,
+    pageIndex: pageIndex,
+    marginTopPixels: marginTopPixels,
+    height: height,
+    bottomPixels: bottomPixels,
+  };
+
+  // Apply scroll
+  parentElement.scrollTo({
+    top: calculateTopPx(objectForCalculateTopPx),
     left: Math.floor(leftPixels),
     behavior: 'smooth',
   });
 }
 
+export function getScrollParent(node: HTMLElement): Nullable<HTMLElement> {
+  if (typeof document === 'undefined') {
+    return null;
+  }
+  if (!node) {
+    return document.documentElement;
+  }
+  if (node.scrollHeight > node.clientHeight) {
+    return node;
+  }
+  return getScrollParent(node.parentElement as HTMLElement);
+}
+
+export function calculateTopPx(calculateTopPxObject: ObjectForCalculateTopPx): number {
+  const { heightWithMargins, pageIndex, marginTopPixels, height, bottomPixels } =
+    calculateTopPxObject;
```

Comment:
I would do this inline. I'd also add units to the variable names.

```suggestion
export function calculateTopPx({

  heightWithMarginsInPx,
  pageIndex,
  marginTopPx,
  heightPx,
  bottomPx,
}: {
  heightWithMarginsInPx: number;
  pageIndex: number;
  marginTopPx: number;
  heightPx: number;
  bottomPx: number;
}): number {
```